kore/src/Kore/Step/SMT/AST.hs

```diff
@@ -322,15 +324,11 @@ type UnresolvedSymbol =
     Symbol SortReference Encodable
 
 encodable :: Kore.Id -> Encodable
-encodable Kore.Id {getId} = Encodable getId
+encodable Kore.Id {getId} = Encodable (Atom getId)
 
-encode :: Encodable -> Text
+encode :: Encodable -> SExpr
 encode (AlreadyEncoded e) = e
-encode (Encodable e) = encodeName e
-
-appendToEncoding :: Encodable -> Text -> Encodable
-appendToEncoding (AlreadyEncoded e) t = AlreadyEncoded (e <> t)
-appendToEncoding (Encodable e) t = Encodable (e <> t)
+encode (Encodable e) = AST.mapSExpr encodeName e
```

Comment:
`mapSExpr` maps `encodeName` to every `Text` from the `SExpression`, but the unit tests would pass even if it mapped only to the `Text` from the first `Atom` (the one that `nameFromSExpr` would return)